extensions/sql-database-projects/src/models/project.ts

```diff
@@ -609,102 +621,23 @@ export class Project implements ISqlProject {
-		}
-
-		// type 2: Project node has Sdk attribute like <Project Sdk="Microsoft.Build.Sql/1.0.0">
-		const sdkAttribute: string = this.projFileXmlDoc!.documentElement.getAttribute(constants.Sdk)!;
-		if (sdkAttribute) {
-			return sdkAttribute.includes(constants.sqlProjectSdk);
-		}
-
-		// type 3: Import node with Sdk attribute like <Import Project="Sdk.targets" Sdk="Microsoft.Build.Sql" Version="1.0.0" />
-		const importNodes = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Import);
-		for (let i = 0; i < importNodes.length; i++) {
-			if (importNodes[i].getAttribute(constants.Sdk) === constants.sqlProjectSdk) {
-				return true;
-			}
-		}
-
-		return false;
-	}
-
 	public async updateProjectForRoundTrip(): Promise<void> {
-		if (this._importedTargets.includes(constants.NetCoreTargets) && !this.containsSSDTOnlySystemDatabaseReferences() // old style project check
-			|| this.isSdkStyleProject) { // new style project check
+		if (this.sqlProjStyle === ProjectType.SdkStyle) {
 			return;
 		}
 
-		TelemetryReporter.sendActionEvent(TelemetryViews.ProjectController, TelemetryActions.updateProjectForRoundtrip);
-
-		if (!this._importedTargets.includes(constants.NetCoreTargets)) {
-			const result = await window.showWarningMessage(constants.updateProjectForRoundTrip(this.projectFileName), constants.yesString, constants.noString);
-			if (result === constants.yesString) {
-				await fs.copyFile(this._projectFilePath, this._projectFilePath + '_backup');
-				await this.updateImportToSupportRoundTrip();
-				await this.updatePackageReferenceInProjFile();
-				await this.updateBeforeBuildTargetInProjFile();
-				await this.updateSystemDatabaseReferencesInProjFile();
-			}
-		} else if (this.containsSSDTOnlySystemDatabaseReferences()) {
-			const result = await window.showWarningMessage(constants.updateProjectDatabaseReferencesForRoundTrip(this.projectFileName), constants.yesString, constants.noString);
-			if (result === constants.yesString) {
-				await fs.copyFile(this._projectFilePath, this._projectFilePath + '_backup');
-				await this.updateSystemDatabaseReferencesInProjFile();
-			}
-		}
-	}
-
-	private async updateImportToSupportRoundTrip(): Promise<void> {
-		// update an SSDT project to include Net core target information
-		for (let i = 0; i < this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Import).length; i++) {
-			const importTarget = this.projFileXmlDoc!.documentElement.getElementsByTagName(constants.Import)[i];
-
-			let condition = importTarget.getAttribute(constants.Condition);
-			let projectAttributeVal = importTarget.getAttribute(constants.Project);
+		// TODO: is this check below still relevant?
+		// if (this._importedTargets.includes(constants.NetCoreTargets) && !this.containsSSDTOnlySystemDatabaseReferences() // old style project check
+		// 	|| this.sqlProjStyle === ProjectType.SdkStyle) { // new style project check
+		// 	return;
+		// }
```

Comment:
@kisantia do you remember what this was supposed check that isn't encapsulated in `sqlProjService.getCrossPlatformCompatibility()`?

Reply:
`this.isCrossPlatformCompatible` should handle what this was checking